packages/hs-react-ui/src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,413 @@
+        ratings.push(
+          <StyledEmptyRank
+            hasStages={hasStages}
+            ratingSelected={isRatingSelected}
+            onClick={clickHandler}
+            id={x}
+            key={x}
+            {...emptyRankProps}
+          >
+            {typeof stages[x] === 'string' && stages[x - 1] !== '' ? (
+              <UnstyledIcon path={stages[x - 1] as string} size="2rem" />
+            ) : (
+              stages[x - 1]
+            )}
+          </StyledEmptyRank>,
+        );
+      }
+    }
+  } else {
+    // No rating stages. Use filled, halfFilled, and empty ratings
+    for (let x = max; x > 0; x--) {
+      if (x <= Math.round(ratingValue - 0.01)) {
+        ratings.push(
+          <StyledFilledRank onClick={clickHandler} id={x} key={x} {...filledRankProps}>
+            {typeof filledRankItem === 'string' && filledRankItem !== '' ? (
+              <UnstyledIcon path={filledRankItem} size="2rem" />
+            ) : (
+              filledRankItem
+            )}
+          </StyledFilledRank>,
+        );
+      } else if (x - 1 === Math.floor(ratingValue) && x - 1 !== ratingValue) {
+        ratings.push(
+          <StyledHalfFilledRank
+            onClick={clickHandler}
+            hasHalfFilledRank={hasHalfFilledRank}
+            id={x}
+            key={x}
+            {...halfFilledRankProps}
+          >
+            {typeof halfFilledRankItem === 'string' && halfFilledRankItem !== '' ? (
+              <UnstyledIcon path={halfFilledRankItem} size="2rem" />
+            ) : (
+              halfFilledRankItem
+            )}
+          </StyledHalfFilledRank>,
+        );
+      } else {
+        ratings.push(
+          <StyledEmptyRank
+            onClick={clickHandler}
+            ratingSelected={isRatingSelected}
+            id={x}
+            key={x}
+            {...emptyRankProps}
+          >
+            {typeof emptyRankItem === 'string' && emptyRankItem !== '' ? (
+              <UnstyledIcon path={emptyRankItem} size="2rem" />
+            ) : (
+              emptyRankItem
+            )}
+          </StyledEmptyRank>,
+        );
+      }
+    }
+  }
+
+  return (
+    <StyledContainer disabled={disabled}>
+      <RatingWrapper
+        onMouseLeave={mouseLeaveHandler}
+        value={ratingValue}
+        {...mergedContainerProps}
+        ref={containerRef}
+      >
+        {ratings.map(ratingItem => ratingItem)}
+      </RatingWrapper>
+      {showDisplay && (
+        <Info>
+          <span style={{ marginRight: '2px', verticalAlign: 'middle' }}>{ratingValue}</span>
+          <span>
+            {isRatingSelectedRef.current ? (
+              <UnstyledIcon style={{ verticalAlign: 'middle' }} path={mdiLockCheck} size="1rem" />
+            ) : (
+              <UnstyledIcon
+                style={{ verticalAlign: 'middle' }}
+                path={mdiLockOpenVariantOutline}
+                size="1rem"
+              />
+            )}
+          </span>{' '}
+        </Info>
```

Comment:
None of these components can be restyled and are not exposed to be restyled or overridden

Reply:
- [ ] Exposed Info container